### ISSUE TITLE
False Positive(E1120) - Changing arguments using a wrapper

### ISSUE THREAD

**Issue by xsnowstorm** [pylint-dev/pylint#9234]
### Bug description

The wrapper changes the function to no longer require an argument.
Pylint does not seem to realize this though.
```python
def wrapper(func):
    def inner():
        func("Hello World")
    return inner

@wrapper
def log(message):
    print(message)

log()
```

### Command used

```shell
pylint test.py --disable=C0114,C0116
```

### Pylint output

```python
************* Module test
test.py:10:0: E1120: No value for argument 'message' in function call (no-value-for-parameter)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 2.86/10, +0.00)
```

### Expected behavior

There should not be a problem, because the program runs normally.

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
```

### OS / Environment

Debian Linux

**Comment by DanielNoord** [comment]
This will (sadly) be impossible to statically infer because there is simply too much code that can happen within `inner()`. Even if we were to fix this very simple case we would start getting requests for more complex cases.
Therefore, I will close this.

**Comment by xsnowstorm** [comment]
okay, thanks for the response
